src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Cancellation.cs

```diff
@@ -536,14 +530,18 @@ public static IEnumerable<object[]> PostAsync_Cancel_CancellationTokenPassedToCo
 
 #if !NETFRAMEWORK
         [OuterLoop("Uses Task.Delay")]
-        [ConditionalTheory]
+        [Theory]
         [MemberData(nameof(PostAsync_Cancel_CancellationTokenPassedToContent_MemberData))]
         public async Task PostAsync_Cancel_CancellationTokenPassedToContent(HttpContent content, CancellationTokenSource cancellationTokenSource)
         {
             if (IsWinHttpHandler)
             {
                 return;
             }
+            if (!TestAsync)
+            {
+                return;
+            }
```

Comment:
Why is this exempted for sync?

Reply:
The same reason as WinHttpHandler (the cancellationToken in readAsync delegate is default).
It internally uses DelegateStream, which calls sync Read (no token on Stream.Read) and the original token gets discarded.

Reply:
Might be worth a comment.